tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/SetInRuleWithArrays.scala

```diff
@@ -90,8 +90,8 @@ class SetInRuleWithArrays(rewriter: SymbStateRewriter) extends SetInRule(rewrite
           nextState = rewriter.rewriteUntilDone(nextState.setRex(tla.appFun(funCell.toNameEx, funsetDomElem.toNameEx)))
           val funAppRes = nextState.asCell
           val powSetDom = nextState.arena.getDom(funsetCdm)
-          val subsetEq = tla.subseteq(funAppRes.toNameEx, powSetDom.toNameEx)
-          nextState = rewriter.rewriteUntilDone(nextState.setRex(subsetEq))
+          val quant = tla.forall(tla.name("x"), funAppRes.toNameEx, tla.in(tla.name("x"), powSetDom.toNameEx))
```

Comment:
Not sure if it's sound to not use a fresh name here?
If not, how would I create one?

Reply:
Renaming in [83bbac4](https://github.com/informalsystems/apalache/pull/1716/commits/83bbac46be1236ee55ee957c89c8269465351204)

Reply:
Ugh, so since we're not inlining nullary LET-INs, rewriting this in Keramelizer actually won't allow us to get rid of this case. Something like the spec in https://github.com/informalsystems/apalache/issues/1735 would still need this rewrite rule.

@konnov Should I still include the rewriting in Keramelizer, even though it's orthogonal to this PR?

Reply:
You are right. Feel free to create two PRs instead of one. We should redesign the preprocessing to make it all work as we expected here.